### PR TITLE
build: use latest Rust toolchain using fenix

### PR DESCRIPTION
**Pull request by ysndr**
Based on @zmitchell's WIP https://github.com/flox/flox/pull/1369.

I picked that up to continue integration of fenix to bring this over the line due despite PTO

### CODE REVIEW

File: flake.nix
Comment (ysndr) [inline flake.nix:149]: @garbas d you remeber why we had to add our own wrapper around rustfmt?
I think it was due to not being able to set rustfmt using the `tools` config but I'm not sure.

Comment (ysndr) [reply flake.nix:149]: indeed since https://github.com/cachix/git-hooks.nix/pull/397 that should be possible without a custom wrapper

File: flake.nix
Comment (mkenigs) [inline flake.nix:31]: nonblocking question: how come we're not doing the follows? Drop the comment?

File: flake.nix
Comment (mkenigs) [inline flake.nix:124]: question nonblocking: how come one uses prev and one uses final?

File: flake.nix
Comment (mkenigs) [inline flake.nix:166]: nonblocking: would be nice to fix clippy warnings

**Review by mkenigs — APPROVED** [review]
Looks good to me but I don't fully understand it, so if you're not confident in the change might be good to get a review from someone else, or I can spend longer on it

File: flake.nix
Comment (ysndr) [reply flake.nix:166]: Are there _new_ warnings?
I can go solve themhere or in a follow up.

File: flake.nix
Comment (ysndr) [reply flake.nix:124]: That was there before, technically it's to avoid infinite recursions but since if the overlay depends on something that itself added. I don't think we need this (it's been this way when I picked it up)

File: flake.nix
Comment (ysndr) [reply flake.nix:31]: We use it as an overlay so setting the follow or not does not change much, we should set it anyway though I think to avoid inflating the lock file.

File: flake.nix
Comment (mkenigs) [reply flake.nix:166]: Yes I presume due to using a different version of clippy

Comment (ysndr) [reply flake.nix:166]: Makes sense new rust = smarter clippy
I'll fix those tomorrow then

Comment (dcarley) [reply flake.nix:166]: Curious: How did you notice the warnings? Should they block CI?

Comment (ysndr) [reply flake.nix:166]: The warnings don't block CI, but they do block pushes locally through git hooks (iff they are in files that were authored on the branch).
other than that, running `cargo clippy` in the `cli` directory will lint the project (speaking of we prob want to add a just target _just_ for that eventually)

File: cli/catalog-api-v1/build.rs
Comment (ysndr) [inline cli/catalog-api-v1/build.rs:1]: this is not strictly in the right PR, but it seems small enough to be slipped in -- if there is greater will i can break that out regardless.

File: flake.nix
Comment (dcarley) [reply flake.nix:166]: Warnings won't block pushes unless we enable `settings.clippy.denyWarnings` which is I guess how you were able to push. I think we should enable it since we didn't have any warnings previously and don't want to introduce any more.

Is there a reason for not running the linting in CI? Presumably @mkenigs had to go out of his way to see the warnings and there's no guarantee that all committers will have the hooks configured, which puts quite a lot of onus on reviewers.

Comment (ysndr) [reply flake.nix:167]: https://github.com/flox/flox/pull/1375

Comment (dcarley) [reply flake.nix:167]: Thanks and thanks ❤️

**Review by dcarley — APPROVED** [review]
Think I was able to follow along 👍🏻 TIL about fenix.

**Pull request merged**
